core/app/models/spree/menu.rb

```diff
@@ -0,0 +1,45 @@
+module Spree
+  class Menu < Spree::Base
+    has_many :menu_items, dependent: :destroy
+    has_many :navigations, inverse_of: :menu
+    has_many :stores, through: :navigations
+
+    before_validation :paremeterize_unique_code
+    after_create :set_root
+
+    validate :uniqueness_of_code_within_stores
```

Comment:
```suggestion
    validate : unique_code, uniqueness: { scope: :store_id }
```

that should do the trick

Reply:
That was my first try: `validates :unique_code, uniqueness: { scope: :store_id }`

But because it is a join table many to many with a through association, it's looking for the `store_id` on the `spree_menus` table, then that exists on the `spree_menus_stores` table.

<img width="943" alt="Screenshot 2021-05-04 at 07 59 26" src="https://user-images.githubusercontent.com/1240766/116969900-f3d45d00-acae-11eb-8760-73e420c43bc5.png">

Do you want me to convert this to a has_one relationship?

Reply:
@MatthewKennedy yes, converting to `belongs_to` is the way to go, this will simplify things in multi-store/cloud scenario a lot

Reply:
@damianlegawiec No problem, I'll sort that.